components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java

```diff
@@ -1396,15 +1397,24 @@ private Cookie getTokenIdCookie(HttpServletRequest req) {
      * @param sessionId            Session Id.
      * @param req                  HttpServlet Request.
      * @param resp                 HttpServlet Response.
+     * @param tenantDomain         Tenant Domain
      * @param loggedInTenantDomain Logged In Tenant Domain.
+     * @param sessionIdentifier    Session Identifier
      */
     private void storeTokenIdCookie(String sessionId, HttpServletRequest req, HttpServletResponse resp,
-                                    String tenantDomain, String loggedInTenantDomain) {
+                                    String tenantDomain, String loggedInTenantDomain, String sessionIdentifier) {
 
         ServletCookie samlssoTokenIdCookie = new ServletCookie(SAML_SSO_TOKEN_ID_COOKIE, sessionId);
         IdentityCookieConfig samlssoTokenIdCookieConfig = IdentityUtil
                 .getIdentityCookieConfig(SAML_SSO_TOKEN_ID_COOKIE);
 
+        // Get age of the samlssoTokenId cookie.
+        SessionContext sessionContext = getSessionContext(sessionIdentifier, loggedInTenantDomain);
+        Integer cookieAge = null;
```

Comment:
```suggestion
        Integer cookieAge = 0;
```

Reply:
We cannot initialize cookieAge to 0. If we initialized to 0, then the cookie will not created as the age is expired. That's why this null check is introduced in https://github.com/wso2-extensions/identity-inbound-auth-saml/pull/341. Also we cannot check if the cookieAge == 0 and ignore as when removing the cookie, we set cookie value to 0 to make it expire.